File: Programas/FrontEnd/src/types/DetalleReserva.ts
import { UnidadAlojamiento } from "./UnidadAlojamiento";

export interface DetalleReserva {
  detalleReservaId?: number;
  reservaId?: number;
  unidadAlojamiento: UnidadAlojamiento;
  precioACobrar: number;
  cantidadPersonas: number;
  fechaDesde: string;
  fechaHasta: string;
}

File: Programas/ApiReservaRes/WebApplication2333/Models/DetalleReserva.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ApiReservaRes.Models
{
    public class DetalleReserva
    {
        public int detalleReservaId { get; set; }

        public int reservaId { get; set; }

        public UnidadAlojamiento unidadAlojamiento { get; set; }
        public decimal precioACobrar { get; set; }
        public int cantidadPersonas { get; set; }

        public DateTime fechaDesde { get; set; }
        public DateTime fechaHasta { get; set; }
    }
}